src/code/audio_synthesis.c

```diff
@@ -1184,27 +1189,42 @@ Acmd* AudioSynth_ProcessEnvelope(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisS
     return cmd;
 }
 
-Acmd* AudioSynth_LoadWaveSamples(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisState* synthState, s32 nSamplesToLoad) {
-    s32 temp_v0;
-    s32 unk6 = noteSubEu->unk_06;
+Acmd* AudioSynth_LoadWaveSamples(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisState* synthState,
+                                 s32 numSamplesToLoad) {
+    s32 numSamplesAvail;
+    s32 harmonicIndexCurAndPrev = noteSubEu->harmonicIndexCurAndPrev;
     s32 samplePosInt = synthState->samplePosInt;
-    s32 repeats;
+    s32 numDuplicates;
 
     if (noteSubEu->bitField1.bookOffset != 0) {
-        AudioSynth_LoadBuffer(cmd++, DMEM_UNCOMPRESSED_NOTE, ALIGN16(nSamplesToLoad * 2), gWaveSamples[8]);
-        gWaveSamples[8] += nSamplesToLoad * 2;
+        // Move the noise wave (that reads compiled assembly as samples) from ram to dmem
+        AudioSynth_LoadBuffer(cmd++, DMEM_UNCOMPRESSED_NOTE, ALIGN16(numSamplesToLoad * sizeof(s16)), gWaveSamples[8]);
+        // Offset the address for the samples read by gWaveSamples[8] to the next set of samples
+        gWaveSamples[8] += numSamplesToLoad * sizeof(s16);
         return cmd;
     } else {
-        aLoadBuffer(cmd++, noteSubEu->samples, DMEM_UNCOMPRESSED_NOTE, 0x80);
-        if (unk6 != 0) {
-            samplePosInt = (samplePosInt * D_801304C0[unk6 >> 2]) / D_801304C0[unk6 & 3];
+        // Move the synthetic wave from ram to dmem
+        aLoadBuffer(cmd++, noteSubEu->waveSampleAddr, DMEM_UNCOMPRESSED_NOTE, WAVE_SAMPLE_COUNT * sizeof(s16));
+
+        // If the harmonic changes, map the offset in the wave from one harmonic to another for continuity
+        if (harmonicIndexCurAndPrev != 0) {
+            samplePosInt = samplePosInt * sNumSamplesPerWavePeriod[harmonicIndexCurAndPrev >> 2] /
+                           sNumSamplesPerWavePeriod[harmonicIndexCurAndPrev & 3];
         }
+
+        // Offset in the 64 samples of gWaveSamples to start processing the wave for continuity
         samplePosInt &= 0x3F;
-        temp_v0 = 0x40 - samplePosInt;
-        if (temp_v0 < nSamplesToLoad) {
-            repeats = ((nSamplesToLoad - temp_v0 + 0x3F) / 0x40);
-            if (repeats != 0) {
-                aDuplicate(cmd++, repeats, DMEM_UNCOMPRESSED_NOTE, DMEM_UNCOMPRESSED_NOTE + 0x80, 0x80);
+        // Number of samples in the initial 64 samples available to be used to process
+        numSamplesAvail = WAVE_SAMPLE_COUNT - samplePosInt;
+
+        // Require duplicates if there are more samples to load than available
+        if (numSamplesToLoad > numSamplesAvail) {
+            // (numSamplesToLoad - numSamplesAvail) is the number of samples needed in duplicates.
+            // Each duplicate copies exactly 64 samples, so divide through WAVE_SAMPLE_COUNT and round-up
```

Comment:
"divide through"? Is that grammatically correct?

Also `64` in comments in this function -> `WAVE_SAMPLE_COUNT` ?

Reply:
64 samples are hardcoded in the microcode for `aDuplicate`, so not really? It's expected that the `WAVE_SAMPLE_COUNT` is the same as the value for `aDuplicate` (This is `aDuplicate`'s only use in the repo afaict). So even having the `WAVE_SAMPLE_COUNT` macro is weird because changing it would break things unless you also update the microcode in `aDuplicate`

And yeah, "divide by" seems much more common than "divide through"